traffic_portal/app/src/common/api/TenantService.js

```diff
@@ -17,53 +17,70 @@
  * under the License.
  */
 
-var TenantService = function(Restangular, messageModel) {
+var TenantService = function($http, ENV, messageModel) {
 
     this.getTenants = function(queryParams) {
-        return Restangular.all('tenants').getList(queryParams);
+        return $http.get(ENV.api['root'] + 'tenants', {params: queryParams}).then(
+            function (result) {
+                return result.data.response;
+            },
+            function (err) {
+                throw err;
+            }
+        )
     };
 
     this.getTenant = function(id) {
-        return Restangular.one("tenants", id).get();
+        return $http.get(ENV.api['root'] + 'tenants', {params: {id: id}}).then(
+            function (result) {
+                return result.data.response[0];
+            },
+            function (err) {
+                throw err;
+            }
+        )
     };
 
     this.createTenant = function(tenant) {
-        return Restangular.service('tenants').post(tenant)
-            .then(
-                function() {
-                    messageModel.setMessages([ { level: 'success', text: 'Tenant created' } ], true);
-                },
-                function(fault) {
-                    messageModel.setMessages(fault.data.alerts, true);
-                }
+        return $http.post(ENV.api['root'] + 'tenants', tenant).then(
+            function(result) {
+                messageModel.setMessages([ { level: 'success', text: 'Tenant created' } ], true);
+                return result;
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, true);
```

Comment:
it should be

`messageModel.setMessages(err.data.alerts, false);`

because failure to create/update/delete will not navigate you to a new state (page) so delay should be false.

Reply:
yeah. I just copied what was there, but you're right - what was there was wrong.